ntrans_gui.py

```diff
@@ -173,25 +188,46 @@ def __init__(self) -> None:
         get_savepath_button.grid(sticky="W", column=1, row=13, padx=(0, 20), pady=(0, 0))
 
         generate_dictionary.grid(column=0, columnspan=2, padx=(0, 0), pady=(0, 0))
+        self.setting_not_defined_warning.grid(column=0, row=15, columnspan=2, padx=(0, 0), pady=(0, 0))
         self.estimated_time_label.grid(sticky="W", column=0, columnspan=2, padx=(30, 0), pady=(0, 0))
 
         about_help_buttonframe.grid(sticky="E", column=0, row=17, columnspan=2, padx=(20, 20), pady=(20, 10))
 
         # Black turn on formatting
         # fmt: on
 
+        self.filepath.trace_add("write", self.update_user_choice_vars)
+        self.target_language_var.trace_add("write", self.update_user_choice_vars)
+        for var in self.checkbox_vars.values():
+            var.trace_add("write", self.update_user_choice_vars)
+
+        self.select_all_var.trace_add("write", self.update_user_choice_vars)
+        self.data_size_var.trace_add("write", self.update_user_choice_vars)
+        for var in self.checkbox_vars.values():
+            var.trace_add("write", self.update_user_choice_vars)
+
         self.select_all_var.trace_add("write", self.update_estimated_time_label)
         self.data_size_var.trace_add("write", self.update_estimated_time_label)
         for var in self.checkbox_vars.values():
             var.trace_add("write", self.update_estimated_time_label)
 
-    def get_save_file_path(self) -> None:
-        savepath = tkinter.filedialog.askdirectory()  # type: ignore
-        if savepath:
-            self.filepath.set(savepath)
+        self.user_choices = {
+            "save_path": self.filepath.get(),
+            "included_ngrams": [
+                n for n, var in self.checkbox_vars.items() if var.get()
+            ],
+            "amount_of_ngrams": self.data_size_var.get(),
+            "target_language": self.target_language_var.get(),
+        }
```

Comment:
This is now repeated twice: it's here in `__init__` and also in `update_user_choice_vars()`. Should `__init__` just call `update_user_choice_vars()`?

Also, is there a reason why this has to be in `self`, instead of just a local variable created just before it is needed?